tests/kernel/wave/wave_e2e_test.py

```diff
@@ -1834,3 +1835,81 @@ def test(
 
     test(input, output)
     assert_close(torch_ref, output, atol=1e-03, rtol=1e-05)
+
+
+@require_e2e
+@pytest.mark.parametrize("shape", [(2, 64)])
+@param_bool("use_buffer_ops", "buf_ops")
+def test_atomic_min(shape, use_buffer_ops, request):
+    run_bench = request.config.getoption("--runperf")
+
+    M = tkl.sym.M
+    N = tkl.sym.N
+    SHMEM_DIM = tkl.sym.SHMEM_DIM
+    SHMEM_IDX = tkl.sym.SHMEM_IDX
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+
+    wave_size = 64
+    BLOCK_M = 2
+    BLOCK_N = 64
+    num_waves = 2
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=wave_size,
+            waves_per_block=(1, 2, 1),
+            vector_shapes={
+                M: int(BLOCK_M / num_waves),
+                N: BLOCK_N,
+                SHMEM_DIM: int(BLOCK_M / num_waves),
```

Comment:
if this is being reduced, this will always be 1. We shouldn't need to introduce a new shmem_dim. Introducing a new shmem_dim would be quite confusing because it doesn't really map to any tensor level dimensions